### PR TITLE
Separate the functions to pcf_flags to an extra module.

### PR DESCRIPTION
**Pull request by guy-maurel**
Ref. #2616

### CODE REVIEW

File: src/uncrustify.cpp [hunk 1]
```diff
@@ -234,8 +235,8 @@ void usage_error(const char *msg)
 static void tease()
 {
    fprintf(stdout,
-           "There are currently %zu options and minimal documentation.\n"
```
Comment (mwoehlke-kitware) [inline src/uncrustify.cpp]: ...this is *solely* because Xenial has a broken mingw compiler??

File: src/pcf_flags.h [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+#ifndef PCF_FLAGS_STR_INCLUDED
+#define PCF_FLAGS_STR_INCLUDED
+
+#include "base_types.h"
+
+#include "enum_flags.h"
+#include "log_levels.h"
+#include "logger.h"
+
+#include <vector>      // do get the definition of size_t
```
Comment (mwoehlke-kitware) [inline src/pcf_flags.h]: BTW, according to https://en.cppreference.com/w/c/types/size_t, `<cstddef>` should also suffice, and is probably lighter.

File: src/pcf_flags.cpp [hunk 3]
```diff
@@ -0,0 +1,107 @@
+/**
+ * @file pcf_flags.cpp
+ *
+ * @author  Guy Maurel
+ * @license GPL v2+
+ */
+
+#define DEFINE_PCF_NAMES
```
Comment (mwoehlke-kitware) [inline src/pcf_flags.cpp]: Does this (and the corresponding `#ifdef` below, which will always be true?) still serve a purpose?

File: src/uncrustify.cpp (hunk 1 above)
Comment (gmaurel) [reply src/uncrustify.cpp]: I don't think

File: src/pcf_flags.cpp (hunk 3 above)
Comment (gmaurel) [reply src/pcf_flags.cpp]: no, thanks

**Comment by gmaurel** [comment]
I lost my branch!
I'll make a new PR

**Pull request closed without merging**